File: Lab5/Infrastructure/Repositories/InMemoryUserRepository.cs
using Lab5.Domain.Entities;
using Lab5.Domain.Interfaces;

namespace Lab5.Infrastructure.Repositories;

public sealed class InMemoryUserRepository : IUserRepository
{
    private readonly IReadOnlyCollection<LibraryUser> _users =
    [
        new StandardUser(Guid.Parse("00000000-0000-0000-0000-000000000201"), "Oksana"),
        new PremiumUser(Guid.Parse("00000000-0000-0000-0000-000000000202"), "Maksym")
    ];

    public LibraryUser? GetById(Guid id) => _users.FirstOrDefault(x => x.Id == id);
}

File: Lab5.Web/src/infrastructure/repositories/InMemoryUserRepository.js
import { StandardUser } from '../../domain/entities/StandardUser.js';
import { PremiumUser } from '../../domain/entities/PremiumUser.js';

export class InMemoryUserRepository {
  constructor() {
    this.users = [
      new StandardUser('u1', 'Оксана'),
      new PremiumUser('u2', 'Максим')
    ];
  }

  getAll() {
    return this.users;
  }

  getById(id) {
    return this.users.find((user) => user.id === id) ?? null;
  }
}
